src/tezos/TezosOperations.ts

```diff
@@ -169,6 +170,32 @@ export namespace TezosOperations {
         }
     }
 
+    export async function appendRevealOperation(
+        network: string,
+        keyStore: KeyStore,
+        fee: number,
+        account: TezosTypes.Account,
+        operations: Operation[]
+    ) {
+        const isManagerKeyRevealed = await isManagerKeyRevealedForAccount(network, keyStore)
+        var returnedOperations: Operation[] = operations;
+        if (!isManagerKeyRevealed) {
+            const revealOp: Operation = {
+                kind: "reveal",
+                source: keyStore.publicKeyHash,
+                fee: fee.toString(),
+                counter: (Number(account.counter) + 1).toString(),
+                gas_limit: '120',
+                storage_limit: '0',
+                public_key: keyStore.publicKey
+            };
+            var operation = operations[0]
+            operation.counter = (Number(account.counter) + 2).toString()
```

Comment:
Looks like you are giving a higher counter to the reveal than the original operation. As discussed previously, the reveal should have a lower counter.

Reply:
Nope! Reveal has a counter 1 less than the new operation (delegation/transaction/origination)

Reply:
Ah, yes!